.github/workflows/production.yml

```diff
@@ -0,0 +1,195 @@
+name: Production
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  FIREBASE_PROJECT_ID: mm-static-site-production
+  FIREBASE_SERVICE_ACCOUNT: "${{ secrets.PRODUCTION_FIREBASE_SERVICE_ACCOUNT }}"
+
+jobs:
+  cache-keys:
+    name: Cache keys
+    runs-on: ubuntu-latest
+
+    outputs:
+      build: build-${{ steps.build.outputs.hash }}-v1
+      node-modules: node-modules-${{ steps.node-modules.outputs.hash }}-v1
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+
+      - id: build
+        run: echo "::set-output name=hash::${{ hashFiles('content/**', 'firebase/**', 'functions/src/**', 'src/**', 'static/**', '**/package.json', '**/yarn.lock') }}"
+
+      - id: node-modules
+        run: echo "::set-output name=hash::${{ hashFiles('**/yarn.lock') }}"
+
+  build:
+    needs: cache-keys
+    name: Build
+    runs-on: ubuntu-latest
+
+    outputs:
+      build-cache-hit: ${{ steps.cache-node-modules.outputs.cache-hit }}
```

Comment:
Shouldn't this be set to `steps.cache-build.outputs.cache-hit`?

Reply:
This output is capturing if the `cache-node-modules` step of the current job used a previous build or not. Perhaps the name of the output could be better? Does the code provide more clarity?

```yml
outputs:
      is-previously-tested: ${{ steps.cache-node-modules.outputs.cache-hit }}
```

and subsequently

```yml
test-a11y:
    if: needs.build.outputs.is-previously-tested != 'true'
```

Reply:
Why test if the node modules cache was hit instead of the build cache?

Reply:
Ah, I see what you're saying now. I was crossed-up and thinking of the `cache-keys` outputs for some reason. Updating this.